core/storage/topic/gae_models.py

```diff
@@ -388,6 +383,15 @@ class TopicSummaryModel(base_models.BaseModel):
     # The thumbnail background color of the topic.
     thumbnail_bg_color = datastore_services.StringProperty(indexed=True)
     version = datastore_services.IntegerProperty(required=True)
+    # A mapping from each story id belonging to the topic to a list of
+    # exploration ids that are linked to the corresponding published story. This
+    # mapping represents the relationships between a topic, its published
+    # stories and the explorations that are linked to each story. This field
+    # must be kept in-sync with updates made to an owned story's explorations,
+    # to the topic's ownership of said stories, and to the publishing or
```

Comment:
Given this, I would expect more calls to be added to generate_topic_summary(). Have you checked that all the necessary hooks to call generate_topic_summary() are in place (and could you provide a list of those hooks with pointers to the relevant code, for verification)?

Reply:
Done. I've provided a list of those hooks within the `topic_services_test.py`.